### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by naanmohammed**
For this milestone I :

- Created a class Classroom .
- Created the has-many/belongs-to relationship between Classroom and  Student.
- Created a class Book.
- Created a class  Rental
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative 'person'
+
+class Student < Person
+  attr_accessor :name
+
+  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
+    super(name, age, parent_permission)
+    @classroom = classroom
+  end
```
Comment (DeliceLydia) [inline student.rb]: - [ ] Please add `classroom` as an `attr_reader`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (DeliceLydia) [inline book.rb]: - [ ] I suggest that you make `rentals` as an `attr_accessor` because we need to read and modify it

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (DeliceLydia) [inline book.rb]: - [ ] To create a rental here we need `person` and `date` and the `book` itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(person, date)
+    Rental.new(date, self, person)
+  end
```
Comment (DeliceLydia) [inline person.rb]: - [ ] Here you should send `date` and `book` and `The person` itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```
def add_rental(book, date)
    Rental.new(book, self, date)
  end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @naanmohammed, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Followed git workflow
- No linter errors
- Created the remaining classes


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @naanmohammed  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**